modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/utils.py

```diff
@@ -13,15 +13,141 @@
 
 """Utilities for internal use by the ``HdkOnNativeDataframe``."""
 
+from typing import Tuple, Union
+from functools import lru_cache
 from collections import OrderedDict
 
 import pandas as pd
+from pandas import Timestamp
 from pandas.core.arrays.arrow.extension_types import ArrowIntervalType
 
 import pyarrow as pa
 from pyarrow.types import is_dictionary
 
 from modin.error_message import ErrorMessage
+from modin.utils import MODIN_UNNAMED_SERIES_LABEL
+
+
+class ColNameCodec:
+    IDX_COL_NAME = "__index__"
+    ROWID_COL_NAME = "__rowid__"
+
+    _RESERVED_NAMES = (MODIN_UNNAMED_SERIES_LABEL, ROWID_COL_NAME)
+    _COL_TYPES = Union[str, int, float, Timestamp, None]
+    _COL_NAME_TYPE = Union[_COL_TYPES, Tuple[_COL_TYPES, ...]]
+
+    def _encode_tuple(values: Tuple[_COL_TYPES, ...]) -> str:  # noqa: GL08
+        dst = ["_T"]
+        count = len(values)
+        for value in values:
+            if isinstance(value, str):
+                dst.append(value.replace("_", "_Q"))
+            else:
+                dst.append(ColNameCodec._ENCODERS[type(value)](value))
+            count -= 1
+            if count != 0:
+                dst.append("_T")
+        return "".join(dst)
+
+    def _decode_tuple(encoded: str) -> Tuple[_COL_TYPES, ...]:  # noqa: GL08
+        items = []
+        for item in encoded[2:].split("_T"):
+            dec = (
+                None
+                if len(item) < 2 or item[0] != "_"
+                else ColNameCodec._DECODERS.get(item[1], None)
+            )
+            items.append(item.replace("_Q", "_") if dec is None else dec(item))
+        return tuple(items)
+
+    _ENCODERS = {
+        tuple: _encode_tuple,
```

Comment:
Isn't that enough to make it work?
```suggestion
        tuple: ColNameCodec._encode_tuple,
```